### PR TITLE
Cached provider should always be used when possible

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://daostack.tpondemand.com/RestUI/Board.aspx#page=board/5209716961861964288&appConfig=eyJhY2lkIjoiQjgzMTMzNDczNzlCMUI5QUE0RUE1NUVEOUQyQzdFNkIifQ==&boardPopup=bug/1648/silent

This completes the second part of the ticket:  "The logic should be the same regardless of how the user chooses to connect / login / make an action (e.g. stake when not connected)…  If we have a cached provider then use it, if no cached provider - show the web3connect dialogue…"

- calls to enableWeb3ProviderAndWarn now call enableWalletProvider instead
- unlike the old enableWeb3ProviderAndWarn, enableWalletProvider always looks to the cache first
- **Further note**  I am proposing here to issue a notification when the user is automatically connected (but not when doing it manually through the Connect or Log In buttons).

### CODE REVIEW

File: src/arc.ts [hunk 1]
```diff
@@ -566,6 +576,77 @@ export function getAccountIsEnabled(): boolean {
   return !!getWeb3Provider();
 }
 
+const ACCOUNTSTORAGEKEY = "currentAddress";
```
Comment (tibetsprague) [inline src/arc.ts]: nice to use underscores between words in constants

File: src/components/Proposal/ActionButton.tsx [hunk 2]
```diff
@@ -68,12 +68,16 @@ class ActionButton extends React.Component<IProps, IState> {
   }
 
   public async handleClickExecute(_event: any): Promise<void> {
-    if (!(await enableWeb3ProviderAndWarn(this.props.showNotification.bind(this)))) { return; }
+    if (!await loadCachedWeb3Provider(this.props.showNotification) &&
```
Comment (tibetsprague) [inline src/components/Proposal/ActionButton.tsx]: if we always use this pattern then shouldn't we have a function that calls and checks both loadCached and enable?

Comment (dkent600) [reply src/components/Proposal/ActionButton.tsx]: yeah, I keep thinking the same thing.  I'll do it.

File: src/arc.ts (hunk 1 above)
Comment (dkent600) [reply src/arc.ts]: Done

File: src/components/Proposal/ActionButton.tsx (hunk 2 above)
Comment (dkent600) [reply src/components/Proposal/ActionButton.tsx]: Done

**Pull request merged**